### ISSUE TITLE
Stats not showing anymore since updating to v3.0.0

### ISSUE THREAD

**Issue by Rhyzak** [Nachtalb/more-upload-stats#8]
I get this when trying to view the stats in web browser after running `/stats`.

![image](https://github.com/user-attachments/assets/4b3407fa-13af-4338-8ef1-00b8edfae379)

The `index.html` file size is 0kb in size and contains nothing inside. I also tried pasting in a backup and renaming it to `stats.json` but it still produces the same bunch of error msgs and a 0kb size `index.html` file.

**Comment by Nachtalb** [comment]
Can you try out this version to see if there is a change https://github.com/Nachtalb/more-upload-stats/actions/runs/10133763319/artifacts/1748444177

I have the feeling it's some kind of encoding problem as you are using mingw64 instead of a native python install

**Comment by Rhyzak** [comment]
I just tried it, I moved over the stats.json, playlist.m3u and index.html, and it now loads fine. But now it overwrites the stats.json file with new data as if starting fresh, the moment someone begins to download from me.
![image](https://github.com/user-attachments/assets/b43881dc-1c81-4149-8b41-b1039146bd26)

The index.html and playlist.m3u files were also not getting updated at all.

I decided to try use the _/up-rebuild_ command to see if that fixes the index.html and playlist.m3u files, it did get them working, but it overwrote the index.html and playlist.m3u files with the data from the new freshly made stats.json file, meaning I can't see any of my previous stats.
![image](https://github.com/user-attachments/assets/34dc3475-7c3b-43eb-bda1-31c9375ab9a8)

Also when I try to back up the file manually from my original 6MB stats.json file:
![image](https://github.com/user-attachments/assets/c3990b9e-0e6b-48b7-8f99-8b04dc34da25)
It only backs up a small stats.json file with the latest files some users are downloading from me since I tried this.
![image](https://github.com/user-attachments/assets/73e53a01-b1c8-49db-9476-cf5c225a155a)

I also tried copying over only the stats.json and playlist.m3u files to see if it would generate an index.html file, but it only created the index.html file with the one user that had just began downloading from me as I was testing this and then deleted all the data in the playlist.m3u file. The stats.json was left alone, but was soon overwritten the moment another user began downloading from me to a fresh new stats.json file. Here is the log for that part:
![image](https://github.com/user-attachments/assets/6f54eb04-078c-45b0-996b-6bfd99082b8c)

**Comment by Rhyzak** [comment]
I also just tried v3.1 and I'm still getting the same issue as I was getting on v3.0.

However, unlike the version you got me to test above, it's updating the stats.json file correctly, and leaving the playlist.m3u file alone. It only appears to be failing to generate the index.html file:
![image](https://github.com/user-attachments/assets/0efb4b66-bd5a-4f54-9297-71f12b5294c2)

**Comment by Nachtalb** [comment]
when you run `/up-backup` or `/up-rebuild` it uses the data in memory, not the data from the file. It loads the file when enabling the plugin or when starting nicotine+. So it makes sense that the `index.html` and the backups don't have teh same data as your 6mb file. The problem here is why I didn't load the `stats.json` at the start 🤔 I'll see if i can find something. 

And yes the 3.1.0 doesn't have the encoding fix in it yet. 

If you want you can send me your `stats.json` for testing. (the full one, not the empty one)

**Comment by Rhyzak** [comment]
Here's the [stats.json](https://github.com/user-attachments/files/16404988/stats.json) file.

**Comment by Nachtalb** [comment]
btw when restarting nicotine with the version I sent you. You should see something like: 
```
... - INFO - Loading statistics from "/path/to/your/stats.json"
```

Is there any more info there, an error or something?

**Comment by Nachtalb** [comment]
your stats loaded no problem on my side hmmmm... 🤔

**Comment by Nachtalb** [comment]
I've uploaded some new changes that give more context when loading the `stats.json`. Please use this version then restart and see if you get any errors. After closing N+ , before starting it again, make sure your `stats.json` is at the right place (insode the `build/` directory of this version of the plugin) : 

https://github.com/Nachtalb/more-upload-stats/actions/runs/10134318776/artifacts/1748535481

**Comment by Rhyzak** [comment]
I just tried doing a fresh install of the plugin, I exited nicotine+, deleted the plugin, then put the one you sent from up top.

> Can you try out this version to see if there is a change https://github.com/Nachtalb/more-upload-stats/actions/runs/10133763319/artifacts/1748444177
> 
> I have the feeling it's some kind of encoding problem as you are using mingw64 instead of a native python install

I copied over the `index.html`, `playlist.m3u` and `stats.json` files to the build folder, loaded nicotine+, and now it all seems to be working fine. I also deleted the `index.html` file to generate a new one with `/stats` and it worked great. I also tried `/up-rebuild`, which also worked correctly at creating the `index.html` and `playlist.m3u` files.

I'm not sure why it wasn't doing it correctly before. But if I recall correctly, I may have placed in the original three files, **after** I had loaded nicotine+. 🤦🏼‍♂️

I have copied the startup log as well incase you still need it:
```
29/07/2024 7:37:20 am Rescanning shares…
29/07/2024 7:37:20 am Loading plugin system
29/07/2024 7:37:20 am Loaded plugin Nicotine+ Commands
29/07/2024 7:37:20 am Loaded plugin Multi Paste
29/07/2024 7:37:20 am Leech Detector: Require users have a minimum of 1 files in 1 shared public folders.
29/07/2024 7:37:20 am Loaded plugin Leech Detector
29/07/2024 7:37:20 am Upload Statistics - DEBUG - Setting up commands
29/07/2024 7:37:20 am Upload Statistics - DEBUG - Command up-backup added <bound method Plugin.backup_cmd of <upload_stats.plugin.Plugin object at 0x0000024a1c77af50>>, <bound method Plugin.backup_cmd of <upload_stats.plugin.Plugin object at 0x0000024a1c77af50>>
29/07/2024 7:37:20 am Upload Statistics - DEBUG - Command up-check-update added <bound method BasePlugin.check_update of <upload_stats.plugin.Plugin object at 0x0000024a1c77af50>>, <bound method BasePlugin.check_update of <upload_stats.plugin.Plugin object at 0x0000024a1c77af50>>
29/07/2024 7:37:20 am Upload Statistics - DEBUG - Command up-list-backups added <bound method Plugin.list_backups of <upload_stats.plugin.Plugin object at 0x0000024a1c77af50>>, <bound method Plugin.list_backups of <upload_stats.plugin.Plugin object at 0x0000024a1c77af50>>
29/07/2024 7:37:20 am Upload Statistics - DEBUG - Command up-open added <bound method Plugin.open of <upload_stats.plugin.Plugin object at 0x0000024a1c77af50>>, <bound method Plugin.open of <upload_stats.plugin.Plugin object at 0x0000024a1c77af50>>
29/07/2024 7:37:20 am Upload Statistics - DEBUG - Command up-open-playlist added <bound method Plugin.open_playlist of <upload_stats.plugin.Plugin object at 0x0000024a1c77af50>>, <bound method Plugin.open_playlist of <upload_stats.plugin.Plugin object at 0x0000024a1c77af50>>
29/07/2024 7:37:20 am Upload Statistics - DEBUG - Command up-rebuild-page added <bound method Plugin.rebuild_page_cmd of <upload_stats.plugin.Plugin object at 0x0000024a1c77af50>>, <bound method Plugin.rebuild_page_cmd of <upload_stats.plugin.Plugin object at 0x0000024a1c77af50>>
29/07/2024 7:37:20 am Upload Statistics - DEBUG - Command up-rebuild-playlist added <bound method Plugin.rebuild_playlist_cmd of <upload_stats.plugin.Plugin object at 0x0000024a1c77af50>>, <bound method Plugin.rebuild_playlist_cmd of <upload_stats.plugin.Plugin object at 0x0000024a1c77af50>>
29/07/2024 7:37:20 am Upload Statistics - DEBUG - Command up-rebuild added <bound method Plugin.rebuild_stats_output_cmd of <upload_stats.plugin.Plugin object at 0x0000024a1c77af50>>, <bound method Plugin.rebuild_stats_output_cmd of <upload_stats.plugin.Plugin object at 0x0000024a1c77af50>>
29/07/2024 7:37:20 am Upload Statistics - DEBUG - Command up-reload added <bound method BasePlugin.reload of <upload_stats.plugin.Plugin object at 0x0000024a1c77af50>>, <bound method BasePlugin.reload of <upload_stats.plugin.Plugin object at 0x0000024a1c77af50>>
29/07/2024 7:37:20 am Upload Statistics - DEBUG - Command up-reset added <bound method Plugin.reset of <upload_stats.plugin.Plugin object at 0x0000024a1c77af50>>, <bound method Plugin.reset of <upload_stats.plugin.Plugin object at 0x0000024a1c77af50>>
29/07/2024 7:37:20 am Upload Statistics - DEBUG - Command up-reset-abort added <bound method Plugin.reset_abort of <upload_stats.plugin.Plugin object at 0x0000024a1c77af50>>, <bound method Plugin.reset_abort of <upload_stats.plugin.Plugin object at 0x0000024a1c77af50>>
29/07/2024 7:37:21 am Upload Statistics - DEBUG - Command up-reset-confirm added <bound method Plugin.reset_confirm of <upload_stats.plugin.Plugin object at 0x0000024a1c77af50>>, <bound method Plugin.reset_confirm of <upload_stats.plugin.Plugin object at 0x0000024a1c77af50>>
29/07/2024 7:37:21 am Upload Statistics - DEBUG - Command up-restore added <bound method Plugin.restore of <upload_stats.plugin.Plugin object at 0x0000024a1c77af50>>, <bound method Plugin.restore of <upload_stats.plugin.Plugin object at 0x0000024a1c77af50>>
29/07/2024 7:37:21 am Upload Statistics - DEBUG - Commands setup complete: {'up-backup': {'callback': <bound method Plugin.backup_cmd of <upload_stats.plugin.Plugin object at 0x0000024a1c77af50>>, 'description': 'Create a backup of the statistics'}, 'up-check-update': {'callback': <bound method BasePlugin.check_update of <upload_stats.plugin.Plugin object at 0x0000024a1c77af50>>, 'description': 'Check for updates'}, 'up-list-backups': {'callback': <bound method Plugin.list_backups of <upload_stats.plugin.Plugin object at 0x0000024a1c77af50>>, 'description': 'List all backups'}, 'up-open': {'callback': <bound method Plugin.open of <upload_stats.plugin.Plugin object at 0x0000024a1c77af50>>, 'description': 'Open the statistics page', 'aliases': ['stats']}, 'up-open-playlist': {'callback': <bound method Plugin.open_playlist of <upload_stats.plugin.Plugin object at 0x0000024a1c77af50>>, 'description': 'Open the top 25 uploads playlist file', 'aliases': ['play']}, 'up-rebuild-page': {'callback': <bound method Plugin.rebuild_page_cmd of <upload_stats.plugin.Plugin object at 0x0000024a1c77af50>>, 'description': 'Rebuild the statistics page\n\n        Args:\n            user_threshold (:obj:`int`, optional): User threshold\n            file_threshold (:obj:`int`, optional): File threshold\n        ', 'parameters': ['[user threshold]', '[file threshold]']}, 'up-rebuild-playlist': {'callback': <bound method Plugin.rebuild_playlist_cmd of <upload_stats.plugin.Plugin object at 0x0000024a1c77af50>>, 'description': 'Rebuild the playlist file'}, 'up-rebuild': {'callback': <bound method Plugin.rebuild_stats_output_cmd of <upload_stats.plugin.Plugin object at 0x0000024a1c77af50>>, 'description': 'Rebuild the statistics page and playlist file\n\n        Args:\n            user_threshold (:obj:`int`, optional): User threshold\n            file_threshold (:obj:`int`, optional): File threshold\n        ', 'parameters': ['[user threshold]', '[file threshold]']}, 'up-reload': {'callback': <bound method BasePlugin.reload of <upload_stats.plugin.Plugin object at 0x0000024a1c77af50>>, 'description': 'Reload the plugin'}, 'up-reset': {'callback': <bound method Plugin.reset of <upload_stats.plugin.Plugin object at 0x0000024a1c77af50>>, 'description': 'Start the reset process'}, 'up-reset-abort': {'callback': <bound method Plugin.reset_abort of <upload_stats.plugin.Plugin object at 0x0000024a1c77af50>>, 'description': 'Cancel the reset process'}, 'up-reset-confirm': {'callback': <bound method Plugin.reset_confirm of <upload_stats.plugin.Plugin object at 0x0000024a1c77af50>>, 'description': 'Create a backup and reset the statistics'}, 'up-restore': {'callback': <bound method Plugin.restore of <upload_stats.plugin.Plugin object at 0x0000024a1c77af50>>, 'description': 'Restore the latest backup\n\n        Args:\n            file (:obj:`str`, optional): Backup file to restore. Default is None.\n        ', 'parameters': ['[backup file]']}}
29/07/2024 7:37:21 am Upload Statistics - DEBUG - Initializing plugin
29/07/2024 7:37:21 am Upload Statistics - INFO - Running version 3.1.1a0
29/07/2024 7:37:21 am Upload Statistics - INFO - Loading statistics from "C:\Users\Kyzak\AppData\Roaming\nicotine\plugins\upload_stats\build\stats.json"
29/07/2024 7:37:21 am Upload Statistics - INFO - Rebuilding statistics page
29/07/2024 7:37:21 am Upload Statistics - DEBUG - Building statistics page
29/07/2024 7:37:21 am Upload Statistics - INFO - Automatic backup triggered
29/07/2024 7:37:21 am Upload Statistics - INFO - Creating a backup for "auto"
29/07/2024 7:37:21 am Upload Statistics - DEBUG - Saving statistics to "C:\Users\Kyzak\AppData\Roaming\nicotine\plugins\upload_stats\build\backups\stats-auto-2024_37_29-07_37_20.json"
29/07/2024 7:37:21 am Loaded plugin Upload Statistics
29/07/2024 7:37:21 am Upload Statistics - DEBUG - Calculating summary
29/07/2024 7:37:21 am Upload Statistics - DEBUG - Created missing file "C:\Users\Kyzak\AppData\Roaming\nicotine\plugins\upload_stats\build\backups\stats-auto-2024_37_29-07_37_20.json"
29/07/2024 7:37:21 am Upload Statistics - DEBUG - Saved statistics to "C:\Users\Kyzak\AppData\Roaming\nicotine\plugins\upload_stats\build\backups\stats-auto-2024_37_29-07_37_20.json"
29/07/2024 7:37:21 am Upload Statistics - INFO - Created a backup at "C:\Users\Kyzak\AppData\Roaming\nicotine\plugins\upload_stats\build\backups\stats-auto-2024_37_29-07_37_20.json"
29/07/2024 7:37:21 am Listening on port: 2242
29/07/2024 7:37:21 am Connecting to server.slsknet.org:2242
29/07/2024 7:37:21 am Connected to server server.slsknet.org:2242, logging in…
29/07/2024 7:37:22 am Upload Statistics - DEBUG - Building user stats with threshold 0
29/07/2024 7:37:23 am Upload Statistics - DEBUG - Building file stats with threshold 0
29/07/2024 7:37:24 am Rescan complete: 5590 folders found
29/07/2024 7:37:25 am UPnP: External port 2242 successfully forwarded to local IP address 192.168.1.124 port 2242
29/07/2024 7:37:37 am 2151 privileged users
29/07/2024 7:37:37 am 24 days, 6 hours, 16 minutes, 11 seconds of Soulseek privileges left
29/07/2024 7:37:38 am Leech Detector: User LuisRossman has no shared files according to the server, requesting shares to verify…
29/07/2024 7:37:38 am Leech Detector: Leecher detected, LuisRossman is only sharing 0 files in 0 folders. Going to message leecher after transfer…
29/07/2024 7:38:04 am Upload Statistics - INFO - Statistics page generated and saved to "C:\Users\Kyzak\AppData\Roaming\nicotine\plugins\upload_stats\build\index.html"
29/07/2024 7:38:04 am Upload Statistics - INFO - Rebuilding playlist with top 25 uploads
29/07/2024 7:38:04 am Upload Statistics - INFO - Playlist generated and saved to "C:\Users\Kyzak\AppData\Roaming\nicotine\plugins\upload_stats\build\playlist.m3u"
```

**Comment by Nachtalb** [comment]
> I'm not sure why it wasn't doing it correctly before. But if I recall correctly, I may have placed in the original three files, after I had loaded nicotine+. 🤦🏼‍♂️

Can happen, no problem 😄 

Thanks to you I also just found some other small issues I was able to fix. 

Maybe I'll take another shot at an auto update feature again some time.

**Comment by Rhyzak** [comment]
> I've uploaded some new changes that give more context when loading the `stats.json`. Please use this version then restart and see if you get any errors. After closing N+ , before starting it again, make sure your `stats.json` is at the right place (insode the `build/` directory of this version of the plugin) :
> 
> https://github.com/Nachtalb/more-upload-stats/actions/runs/10134318776/artifacts/1748535481

Everything is working great on this version as well. 👍

**Comment by Nachtalb** [comment]
I released the package as 3.1.1 with the latest bug fixes including the one from here: https://github.com/Nachtalb/more-upload-stats/releases/tag/v3.1.1
